### ISSUE TITLE
No warning on unsupported methods missing object name

### ISSUE THREAD

**Issue by gabriele-cannizzaro** [amilajack/eslint-plugin-compat#395]
It looks like the plugin can't identify the unsupported use of a method call that doesn't include the class name:

```javascript
'string'.repeat(5); // false negative
new String().repeat(5); // ! ESLint: String.repeat() is not supported in IE 11 (compat/compat)
```

In this case the first use of repeat() is not flagged as incompatible with IE, while the second one is. Is there a way to flag any use of incompatible methods on string literals?

---

**EDIT**: changed the above to reflect the much clearer rewrite by @HansBrende in the [comment below](https://github.com/amilajack/eslint-plugin-compat/issues/395#issuecomment-1010024924). Original:

~~It looks like the plugin can't identify the unsupported use of a method call that doesn't include the object name:~~

![image](https://user-images.githubusercontent.com/7521652/90979841-926cc380-e54f-11ea-8d36-42d077a4a190.png)

~~In this case the first use of repeat() is not flagged as incompatible with IE, while the second one is. Is there a way to flag any use of incompatible methods?~~

**Comment by diegocr** [comment]
Same here...someone at my team used the es2021's `replaceAll()` method and there was no warning thrown from ESLint, the code ended up in production and disaster happened...

Moreover, i have just tried that `String.repeat(5)` here and i am not even getting a warning, nothing, which is very odd...using latest public releases of every npm package at time of writing. 🤔 

May i ask what's your config @gabriele-cannizzaro ? Here's my [browserlist](https://github.com/meganz/webclient/blob/8ebbd94ead0b4d58740a2b0c0389bc1f66bf3e9f/package.json#L31-L36) and [config](https://github.com/meganz/webclient/blob/8ebbd94ead0b4d58740a2b0c0389bc1f66bf3e9f/.eslintrc.json#L155)

**Comment by yoyo837** [comment]
Same with @diegocr in production. So sad.

**Comment by ljharb** [comment]
`String.repeat` is not a thing that exists, so it would be just as weird for something to lint against it as to lint against `String.yogurt`.

**Comment by HansBrende** [comment]
@gabriele-cannizzaro hmm, I'm gonna go out on a limb here and guess that by "`String.repeat` is not a thing that exists", he means **"`String.repeat` is not a thing that exists"**.
<img width="685" alt="Screen Shot 2022-01-11 at 7 06 42 AM" src="https://user-images.githubusercontent.com/5846890/148948387-d0b321da-6f6f-4abe-ab30-256013a24b8a.png">

**Comment by HansBrende** [comment]
@ljharb for clarity, let's _pretend_ OP said:

> It looks like the plugin can't identify the unsupported use of a method call that doesn't include the object name:
```javascript
'string'.repeat(5); // false negative
new String().repeat(5); // ! ESLint: String.repeat() is not supported in IE 11(compat/compat)
```
> In this case the first use of repeat() is not flagged as incompatible with IE, while the second one is. Is there a way to flag any use of incompatible methods?

Now the issue at hand is clear.

**Comment by gabriele-cannizzaro** [comment]
@HansBrende thank you for the rewrite, that makes it indeed much clearer. Re-reading my original post and the thread I was obviously confusing things :-)
